app/lib/surface/Surface.coffee

```diff
@@ -519,7 +519,16 @@ module.exports = Surface = class Surface extends CocoClass
     @setPlayingCalled = false  # Don't overwrite playing settings if they changed by, say, scripts.
     @frameBeforeCast = @currentFrame
     # This is where I wanted to trigger a rewind, but it turned out to be pretty complicated, since the new world gets updated everywhere, and you don't want to rewind through that.
-    @setProgress 0, 0 unless @options.level?.get('product') is 'codecombat-junior'
+    shouldRestart = @options.level?.get('product') isnt 'codecombat-junior'
```

Comment:
shouldn't be `@options.level?.get('product') is 'codecombat-junior'` and use that instead of doing a `not shouldRestart`. Or is there a reason to have negation here and do negation in `if` condition

Reply:
I guess you could go either way. It'd be weird to name the condition `shouldntRestart` or something.